Conscrypt error on KitKat and lower

**Affected app**
Name: Google Calendar Sync 5.2.3-99827563 (2015080710)
Package id: com.google.android.syncadapters.calendar

**Describe the bug**
It was working in the past but now no longer works.

**System**
Android Version: Android 4.4.4

**microG**
microG Core version: 0.2.28.231657-28 (a749fe4)
microG Self-Check results: All ticked
![self-check](https://github.com/microg/GmsCore/assets/15793015/b613bd58-036c-4028-a644-328c19700c84)

**Additional context**
Logcat:
```
I/ActivityManager(  481): Start proc com.google.android.syncadapters.calendar for service com.google.android.syncadapters.calendar/.CalendarSyncAdapterService: pid=1561 uid=10054 gids={50054, 3003}
D/dalvikvm(  481): GC_CONCURRENT freed 11507K, 68% free 6511K/19892K, paused 1ms+1ms, total 24ms
W/audio_hw_primary(  166): out_write() limiting sleep time 44149 to 23219
W/audio_hw_primary(  166): out_write() limiting sleep time 55759 to 23219
D/GmsServicesProvider(  839): query caller=com.google.android.syncadapters.calendar name=calendar_ssa_analytics_sampling_rate value=null
I/dalvikvm( 1561): Could not find method android.app.Notification$Builder.setLocalOnly, referenced from method com.google.android.gms.common.GooglePlayServicesUtil.zza
W/dalvikvm( 1561): VFY: unable to resolve virtual method 25: Landroid/app/Notification$Builder;.setLocalOnly (Z)Landroid/app/Notification$Builder;
D/dalvikvm( 1561): VFY: replacing opcode 0x6e at 0x00c2
I/dalvikvm( 1561): Could not find method android.content.pm.PackageManager.getPackageInstaller, referenced from method com.google.android.gms.common.GooglePlayServicesUtil.zzl
W/dalvikvm( 1561): VFY: unable to resolve virtual method 174: Landroid/content/pm/PackageManager;.getPackageInstaller ()Landroid/content/pm/PackageInstaller;
D/dalvikvm( 1561): VFY: replacing opcode 0x6e at 0x000b
W/audio_hw_primary(  166): out_write() limiting sleep time 37369 to 23219
D/GmsPlayLogSvc(  839): onBind: Intent { act=com.google.android.gms.playlog.service.START pkg=com.google.android.gms }
D/dalvikvm( 1561): GC_CONCURRENT freed 277K, 11% free 2779K/3120K, paused 3ms+1ms, total 7ms
D/GmsPlayLogSvc(  839): bound by: GetServiceRequest{serviceId=PLAY_LOG, gmsVersion=7571000, packageName='com.google.android.syncadapters.calendar', extras=Bundle[{}]}
W/audio_hw_primary(  166): out_write() limiting sleep time 28979 to 23219
I/AnalyticsLogBase( 1561): PlayLogger.onLoggerConnected
I/dalvikvm( 1561): Could not find method android.content.Context.checkSelfPermission, referenced from method com.google.android.calendar.PermissionsCompat.checkSelfPermission
W/dalvikvm( 1561): VFY: unable to resolve virtual method 97: Landroid/content/Context;.checkSelfPermission (Ljava/lang/String;)I
D/dalvikvm( 1561): VFY: replacing opcode 0x6e at 0x0006
D/GmsServicesProvider(  839): query caller=com.google.android.syncadapters.calendar name=google_calendar_calendar_colors value=null
D/GmsServicesProvider(  839): query caller=com.google.android.syncadapters.calendar name=google_calendar_event_colors value=null
E/ProviderInstaller( 1561): Failed to install provider: Didn't find class "com.google.android.gms.common.security.ProviderInstallerImpl" on path: DexPathList[[zip file "/system/framework/com.android.location.provider.jar", zip file "/system/priv-app/GmsCore.apk"],nativeLibraryDirectories=[/system/lib]]
E/ConscryptUtils( 1561): Unrecoverable error from installIfNeeded, in installSecurityProvider
E/ConscryptUtils( 1561): com.google.android.gms.common.GooglePlayServicesNotAvailableException
E/ConscryptUtils( 1561): 	at com.google.android.gms.security.ProviderInstaller.installIfNeeded(Unknown Source)
E/ConscryptUtils( 1561): 	at com.google.android.syncadapters.calendar.ConscryptUtils.installSecurityProvider(ConscryptUtils.java:21)
E/ConscryptUtils( 1561): 	at com.google.android.syncadapters.calendar.CalendarSyncAdapterApiary.performSync(CalendarSyncAdapterApiary.java:536)
E/ConscryptUtils( 1561): 	at com.google.android.syncadapters.calendar.CalendarSyncAdapterApiary.onPerformLoggedSync(CalendarSyncAdapterApiary.java:440)
E/ConscryptUtils( 1561): 	at com.android.emailcommon.syncadapter.LoggingThreadedSyncAdapter.onPerformSync(LoggingThreadedSyncAdapter.java:49)
E/ConscryptUtils( 1561): 	at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:259)
E/CalendarSyncAdapter( 1561): Conscrypt error during sync
E/CalendarSyncAdapter( 1561): com.google.android.syncadapters.calendar.ConscryptInstallationException: Error from installSecurityProvider
E/CalendarSyncAdapter( 1561): 	at com.google.android.syncadapters.calendar.ConscryptUtils.installSecurityProvider(ConscryptUtils.java:36)
E/CalendarSyncAdapter( 1561): 	at com.google.android.syncadapters.calendar.CalendarSyncAdapterApiary.performSync(CalendarSyncAdapterApiary.java:536)
E/CalendarSyncAdapter( 1561): 	at com.google.android.syncadapters.calendar.CalendarSyncAdapterApiary.onPerformLoggedSync(CalendarSyncAdapterApiary.java:440)
E/CalendarSyncAdapter( 1561): 	at com.android.emailcommon.syncadapter.LoggingThreadedSyncAdapter.onPerformSync(LoggingThreadedSyncAdapter.java:49)
E/CalendarSyncAdapter( 1561): 	at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:259)
E/CalendarSyncAdapter( 1561): Caused by: com.google.android.gms.common.GooglePlayServicesNotAvailableException
E/CalendarSyncAdapter( 1561): 	at com.google.android.gms.security.ProviderInstaller.installIfNeeded(Unknown Source)
E/CalendarSyncAdapter( 1561): 	at com.google.android.syncadapters.calendar.ConscryptUtils.installSecurityProvider(ConscryptUtils.java:21)
E/CalendarSyncAdapter( 1561): 	... 4 more
W/audio_hw_primary(  166): out_write() limiting sleep time 31178 to 23219
I/ActivityManager(  481): Killing 933:com.android.dialer/u0a4 (adj 15): empty #17
D/SyncManager(  481): failed sync operation [sample_mail]@gmail.com u0 (com.google), com.android.calendar, SERVER, latestRunTime 115743, reason: AutoSync, SyncResult: stats [ numParseExceptions: 1]
D/SyncManager(  481): not retrying sync operation because the error is a hard error: [sample_mail]@gmail.com u0 (com.google), com.android.calendar, SERVER, latestRunTime 115893, reason: AutoSync
D/dalvikvm(  633): GC_CONCURRENT freed 424K, 11% free 3988K/4464K, paused 2ms+2ms, total 9ms
I/GAV2    ( 1561): Thread[GAThread,5,main]: No campaign data found.
I/LeapdroidClipboardServer(  481): accept new connection
I/LeapdroidClipboardServer(  481): handle new connection in new thread
I/LeapdroidClipboardServer(  481): new connection is 99
I/LeapdroidClipboardServer(  481): read 19 UTF-16 chars
I/LeapdroidClipboardServer(  481): Client Says: is not an ABI2 app
I/LeapdroidClipboardClient(  481): copy plain text
I/LeapdroidClipboardClient(  481): connected leapdroid.clipboard.server.c
I/LeapdroidClipboardClient(  481): the message is: is not an ABI2 app
```

--- a/play-services-core/multidex-keep.pro
+++ b/play-services-core/multidex-keep.pro
@@ -1,5 +1,9 @@
-# Make sure maps is in primary dex file
+# Make sure maps is in the primary dex file
 -keep class com.google.android.gms.maps.** { *; }
 -keep class org.microg.gms.maps.** { *; }
 -keep class com.mapbox.** { *; }
 -keep class org.oscim.** { *; }
+
+# Make sure these classes are in the primary dex file otherwise Conscrypt will error out on legacy Android versions
+-keep class com.google.android.gms.common.security.ProviderInstallerImpl { *; }
+-keep class com.google.android.gms.org.conscrypt.** { *; }

--- a/play-services-core/multidex-keep.pro
+++ b/play-services-core/multidex-keep.pro
@@ -4,6 +4,9 @@
 -keep class com.mapbox.** { *; }
 -keep class org.oscim.** { *; }
 
-# Make sure these classes are in the primary dex file otherwise Conscrypt will error out on legacy Android versions
+# Keep Dynamite Loader in the primary dex file otherwise it will error out on legacy Android versions
+-keep class com.google.android.gms.chimera.container.DynamiteLoaderImpl { *; }
+
+# Keep Conscrypt in the primary dex file otherwise it will error out on legacy Android versions
 -keep class com.google.android.gms.common.security.ProviderInstallerImpl { *; }
 -keep class com.google.android.gms.org.conscrypt.** { *; }

